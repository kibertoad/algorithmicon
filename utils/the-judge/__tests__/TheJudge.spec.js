const { TheJudge } = require('../TheJudge')

describe('The Judge', () => {
  describe('judgeByFile', () => {
    it('accepts correct answer', () => {
      const judge = new TheJudge({
        baseDir: __dirname
      })
      const implementationFn = (inputLines) => {
        return inputLines.map((inputLine) => {
          const columns = inputLine.split(' ')
          return Number.parseInt(columns[0]) + Number.parseInt(columns[1])
        })
      }

      return judge.judgeByFiles('happy-input', 'happy-output', implementationFn).then((result) => {
        expect(result.errors.length).toEqual(0)
      })
    })

    it('rejects incorrect answers', () => {
      const judge = new TheJudge({
        baseDir: __dirname
      })
      const implementationFn = (inputLines) => {
        return inputLines.map((inputLine) => {
          const columns = inputLine.split(' ')
          return Number.parseInt(columns[0]) + Number.parseInt(columns[1])
        })
      }

      return judge
        .judgeByFiles('unhappy-input', 'unhappy-output', implementationFn)
        .then((result) => {
          expect(result.errors.length).toEqual(2)
          expect(result.errors).toMatchSnapshot()
        })
    })
  })
})
