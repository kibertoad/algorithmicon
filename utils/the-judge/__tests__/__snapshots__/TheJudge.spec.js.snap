// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Judge judgeByFile rejects incorrect answers 1`] = `
Array [
  "Result mismatch for input case_2.txt. Expected: 4,1,4,0,1099, actual 4,13,4,0,1099",
  "Result mismatch for input case_3.txt. Expected: 4,13,4,1,1099, actual 4,13,4,0,1099",
]
`;
